src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -4983,7 +5002,19 @@ public override bool PreVisit(CallNode node)
                 _currentScope = scopeNew.Parent;
                 PostVisit(node.Args);
 
-                // Typecheck the invocation.
+                if (maybeFunc.HasIdentifiers && _features.HasFlag(Features.SupportIdentifiers))
+                {
+                    foreach (var arg in args.Select((a, i) => a is FirstNameNode firstNameNode && maybeFunc.IsIdentifierParam(i) ? (i, firstNameNode) : (-1, null)).Where(a => a.i != -1))
```

Comment:
>  foreach (var arg in args.Select((a, i) => a is FirstNameNode firstNameNode && maybeFunc.IsIdentifierParam(i) ? (i, firstNameNode) : (-1, null)).Where(a => a.i != -1)) [](http://example.com/codeflow?start=18&length=168)

This expression is too complicated.
can you unroll the for-loop, and just avoid the linq expression and ?: operator.